packages/react/src/components/form/Radio/useRadio.ts

```diff
@@ -0,0 +1,50 @@
+import { type ChangeEvent, type ReactNode, useId, useState } from 'react';
+
+export type UseRadioProps = {
+  disabled?: boolean;
+  error?: ReactNode;
+  name?: string;
+  readOnly?: boolean;
+  value?: string;
+  onChange?: (
+    nextValue: string,
+    currentValue: string,
+    event: ChangeEvent<HTMLInputElement>,
+  ) => void;
+};
+
+export function useRadio({
+  error,
+  name,
+  onChange,
+  value = '',
+  ...rest
+}: UseRadioProps) {
+  const [currentValue, setValue] = useState(value);
+  const nameFallback = useId();
+  const errorId = useId();
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const nextValue = event.target.value;
+    setValue(nextValue);
+    onChange?.(nextValue, currentValue, event);
+  };
+
+  return {
+    value: currentValue,
+    setValue,
+    getProps: (radioValue: string) => ({
+      'aria-describedby': error ? errorId : undefined,
+      'aria-invalid': !!error,
+      checked: currentValue === radioValue,
+      name: name || nameFallback,
+      onChange: handleChange,
+      value: radioValue,
```

Comment:
suggest calling this `getRadioProps` to be consistent with `validationMessageProps`.

nitpick, `radioValue` could also just be called `value`?

Reply:
Can do ☺️